ee/tuf/autoupdate.go

```diff
@@ -233,6 +256,62 @@ func (ta *TufAutoupdater) Interrupt(_ error) {
 	ta.interrupt <- struct{}{}
 }
 
+// Do satisfies the actionqueue.actor interface; it allows the control server to send
+// requests down to autoupdate immediately.
+func (ta *TufAutoupdater) Do(data io.Reader) error {
+	var updateRequest controlServerAutoupdateRequest
+	if err := json.NewDecoder(data).Decode(&updateRequest); err != nil {
+		ta.slogger.Log(context.TODO(), slog.LevelWarn,
+			"received update request in unexpected format from control server, discarding",
+			"err", err,
+		)
+		// We don't return an error because we don't want the actionqueue to retry this request
+		return nil
+	}
+
+	binariesToUpdate := make([]autoupdatableBinary, 0)
+	for _, b := range updateRequest.BinariesToUpdate {
+		if val, ok := autoupdatableBinaryMap[b.Name]; ok {
```

Comment:
You could invert this, and use https://pkg.go.dev/slices#DeleteFunc to remove non matches from `updateRequest`

Reply:
I've got to build a new slice from `updateRequest` anyway, might as well leave the request unmodified